lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -82,6 +121,52 @@ def self.serialize(store:, name:, serializer: nil)
         store_accessor store, name
       end
 
+      def buffers
+        pm_storage_adapter.buffers
+      end
+
+      # TODO: support databases that dont support upserts(pg 9.4, etc)
+      def self.create_later(attrs, storage_adapter)
+        element = new(attrs)
+        storage_adapter.buffers[:upsert][element.unique_identifier] = element
+      end
+
+      # NB: delete_all in AR bypasses relation logic, which shouldn't matter here.
+      def self.bulk_destroy(elements)
+        id_groups = elements.reduce(Hash.new { |h,k| h[k] = [] }) do |memo,(_,el)|
+          if el.is_a?(UserAttribute) || el.is_a?(ObjectAttribute)
+            memo[el.class] << el.id
+          end
+
+          memo
+        end
+
+        PolicyElement.where(unique_identifier: elements.keys).delete_all
```

Comment:
@CGDS @HonoreDB Don't these `delete_all`s mean that we won't get callbacks for Archon notifications or audits? Don't we want those?